### PR TITLE
Convert fetches to function calls for events page

### PR DESCRIPTION
**Pull request by dan-mba**
## Fixes Issue
Fixes events page for #5069 

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/5152"></a>

### CODE REVIEW

File: pages/api/events.js [hunk 1]
```diff
@@ -53,5 +67,10 @@ export default async function handler(req, res) {
     .filter((event) => new Date(event.date.end) > new Date())
     .sort((a, b) => new Date(a.date.start) - new Date(b.date.start));
 
-  res.status(200).json(eventsFiltered);
+  return {
+    statusCode: 200,
+    events: {
+      eventsArray: eventsFiltered
+    }
```
Comment (eddiejaoude) [inline pages/api/events.js:74]: Having this nested with the type I don't think is consistent with similar parts of the code?

Comment (dan-mba) [reply pages/api/events.js:74]: The problem is this is the only one I have updated so far that didn't return an object.

In order to have a place to put an error message, I return an object.
The object contains the array on success or the error message on failure

Comment (eddiejaoude) [reply pages/api/events.js:74]: ah ok, what about having the error message outside the data, similar to `status`?

Comment (dan-mba) [reply pages/api/events.js:74]: Doing it that way would make it more difficult to work with the existing API. 

It make more sense for me for an API to return a status code & JSON payload.
This puts the responsibility on the caller to check the status code & evaluate the JSON payload depending on that status.

**Review by eddiejaoude — APPROVED** [review]
Looks good, thank you 👍

**Pull request merged**